sycl/test-e2e/syclcompat/device/device.cpp

```diff
@@ -159,12 +159,26 @@ void test_device_ext_api() {
   auto major = dev_.get_major_version();
   test_major_version(dev_, major);
   dev_.get_minor_version();
-  dev_.get_max_compute_units();
   dev_.get_max_clock_frequency();
   dev_.get_integrated();
-  syclcompat::device_info Info;
-  dev_.get_device_info(Info);
-  Info = dev_.get_device_info();
+
+  int max_cu = dev_.get_max_compute_units();
+  int max_wg_size = dev_.get_max_work_group_size();
+  size_t global_mem_size = dev_.get_global_mem_size();
+  {
+    syclcompat::device_info Info;
+    dev_.get_device_info(Info);
+    assert(Info.get_max_compute_units() == max_cu);
+    assert(Info.get_max_work_group_size() == max_wg_size);
+    assert(Info.get_global_mem_size() == global_mem_size);
+  }
+  {
+    syclcompat::device_info Info = dev_.get_device_info();
+    assert(Info.get_max_compute_units() == max_cu);
+    assert(Info.get_max_work_group_size() == max_wg_size);
+    assert(Info.get_global_mem_size() == global_mem_size);
+  }
```

Comment:
Can we improve the test a little bit to ensure the cached path is actually taken?

Reply:
Can't seem to find any way to improve it at this level (device_info is returned by value).